README.md

```diff
@@ -57,5 +57,243 @@ External code not found - this code is not associated with any sold ticket in La
+```
+DELETE /discount-code/discount_1/event/event_50/item/item_1
+DELETE /discount-code/discount_1/event/event_50/item/item_2
+```
+What is the expected result of ? 
+
+```
+LIST /discount-code/discount_1/events
+GET /discount-code/discount_1/event/event_50
+```
+
+In the API the result will be empty, in the Admin app UX, the result is actually 1, because the UX is adding a record for discount_1 - event_50 when all items are removed for an event
+```
+PUT /discount-code/discount_1/event/event_50
+```
+
+### List 
+```
+GET /discount-code/list/event/<EVENT_ID>?api_key=<YOUR_API_KEY>
+```
+Response sample
+```
+{
+    "stt": "OK",
+    "message": {
+        "event_id": "<EVENT_ID>",
+        "text": "Discount codes list.",
+        "discount_codes": "[
+            {\"discount_id\":12345,\"events_id\":500,\"item_id\":null,\"code\":\"SUPER\",\"type\":\"value\",\"value\":\"5.0000\",\"active\":\"yes\"},
+            {\"discount_id\":54321,\"events_id\":500,\"item_id\":10,\"code\":\"MEGA\",\"type\":\"percent\",\"value\":\"5.0000\",\"active\":\"yes\"},
+            {\"discount_id\":54321,\"events_id\":500,\"item_id\":20,\"code\":\"2020\",\"type\":\"percent\",\"value\":\"5.0000\",\"active\":\"no\"}
+        ]"
+    }
+}
+```
+
+### Create
+Create a discount code for a specific event and items. 
+```
+POST /discount-code/create?api_key=<YOUR_API_KEY>
+```
+Input parameters
+- **code(*)** : string (max length 20) (*) - The discount code name, unique across L2T.
+- **type(*)** : enum[percent, value] - The discount type, can be a value (e.g. 5€) or a percentage (e.g. 10%).
+- **value(*)** : number (max length 8) - Actual discount value (e.g. 5).
+- **limit** : number - Number of times this code can be used (e.g. 100 - after 100 usages in checkout the discount will no longer take effect).
+- **expires** : date (format - yyyy-mm-dd) - Date until when the discount is valid (e.g. "2023-12-21").
+- **active** : boolean - If the discount is active or inactive.
+- **event_id(*)** : number - The event to which this discount will apply.
+- **item_id** : array[number] - Specific items inside an event (e.g. [10, 20], the discount will apply only to event item 10 and 20). In no items are provided, the discount applies to the whole event_id.
+
+**(*)** - required
+
+
+Payload sample
+```
+{
+    "code": "MEGA",
+    "value": 5,
+    "type": "percent",
+    "limit": 100,
+    "expires": "2023-12-21",
+    "active": true,
+    "event_id": 500,
+    "item_id": [10, 20]
+}
+```
+
+### Activate
+Activate a discount code.
+```
+PUT /discount-code/<DISCOUNT_ID>/activate?api_key=<YOUR_API_KEY>
+```
+Response
+```
+"message": {
+    "discount_id": "DISCOUNT_ID",
+    "text": "Deactivate discount: DISCOUNT_ID",
+    "active": "yes"
+},
+```
+
+### Deactivate
+Deactivate a discount code.
+```
+PUT /discount-code/<DISCOUNT_ID>/deactivate?api_key=<YOUR_API_KEY>
+```
+Reponse
+```
+"message": {
+    "discount_id": "DISCOUNT_ID",
+    "text": "Activated discount: DISCOUNT_ID",
```

Comment:
I think deactivate and activate are switched :)

Reply:
changed!